framework/src/main/java/org/tron/core/metrics/blockchain/BlockChainMetricManager.java

```diff
@@ -1,99 +1,172 @@
+                  1L);
         }
       }
     }
-  }
 
-  public List<WitnessInfo> getNoUpgradedSRList() {
-    List<WitnessInfo> noUpgradedWitness = new ArrayList<>();
+    //TPS
+    if (block.getTransactions().size() > 0) {
+      metricsService.meterMark(MetricsKey.BLOCKCHAIN_TPS, block.getTransactions().size());
+    }
+  }
 
-    List<ByteString> address = chainBaseManager.getWitnessScheduleStore().getActiveWitnesses();
-    for (ByteString it : address) {
-      if (witnessVersion.containsKey(it.toStringUtf8())
-          && witnessVersion.get(it.toStringUtf8()).getVersion() != currentVersion) {
-        WitnessInfo witness = witnessVersion.get(it.toStringUtf8());
-        noUpgradedWitness.add(witness);
+  private List<WitnessInfo> getSrList() {
+    List<WitnessInfo> witnessInfos = new ArrayList<>();
+
+    List<WitnessCapsule> witnessCapsuleList = chainBaseManager.getWitnessStore().getAllWitnesses();
+    for (WitnessCapsule witnessCapsule : witnessCapsuleList) {
+      if (witnessCapsule.getIsJobs()) {
+        String address = Hex.toHexString(witnessCapsule.getAddress().toByteArray());
+        if (witnessInfo.containsKey(address)) {
+          BlockHeader blockHeader = witnessInfo.get(address);
+          WitnessInfo witness = new WitnessInfo(address, blockHeader.getRawData().getVersion());
+          witnessInfos.add(witness);
+        }
       }
     }
-    return noUpgradedWitness;
+    return witnessInfos;
   }
 
-  public RateInfo getBlockProcessTime() {
+  private RateInfo getBlockProcessTime() {
     RateInfo blockProcessTime = new RateInfo();
-
+    blockProcessTime.setCount(metricsService.getMeter(MetricsKey.BLOCKCHAIN_BLOCK_COUNT)
+            .getCount());
     blockProcessTime.setMeanRate(getAvgBlockProcessTimeByGap(0));
     blockProcessTime.setOneMinuteRate(getAvgBlockProcessTimeByGap(1));
     blockProcessTime.setFiveMinuteRate(getAvgBlockProcessTimeByGap(5));
     blockProcessTime.setFifteenMinuteRate(getAvgBlockProcessTimeByGap(15));
     return blockProcessTime;
   }
 
-  public RateInfo getTransactionRate() {
+  private RateInfo getTransactionRate() {
     Meter transactionRate = metricsService.getMeter(MetricsKey.BLOCKCHAIN_TPS);
     RateInfo tpsInfo = new RateInfo();
+    tpsInfo.setCount(transactionRate.getCount());
     tpsInfo.setMeanRate(transactionRate.getMeanRate());
     tpsInfo.setOneMinuteRate(transactionRate.getOneMinuteRate());
     tpsInfo.setFiveMinuteRate(transactionRate.getFiveMinuteRate());
     tpsInfo.setFifteenMinuteRate(transactionRate.getFifteenMinuteRate());
     return tpsInfo;
   }
 
-  public List<WitnessInfo> getNoUpgradedSR() {
+  private List<WitnessInfo> getNoUpgradedSR() {
     List<WitnessInfo> witnesses = new ArrayList<>();
-    for (WitnessInfo it : getNoUpgradedSRList()) {
-      WitnessInfo noUpgradeSR = new WitnessInfo();
-      noUpgradeSR.setAddress(it.getAddress());
-      noUpgradeSR.setVersion(it.getVersion());
+    for (WitnessInfo it : getSrList()) {
```

Comment:
this is only SR list, not no upgraded SR list.
Need to check current version with max version